src/CheckStrings.jl

```diff
@@ -0,0 +1,200 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+## Functions to check validity of UTF-8, UTF-16, and UTF-32 encoded strings,
+#  and also to return information necessary to convert to other encodings
+
+module CheckStrings
+
+using Encodings
+
+export check_string
```

Comment:
Ideally in the future, we'd have a `Strings` module in Base and we could just have `Strings.check()` and leave it unexported.